### ISSUE TITLE
Unexpected alsoValidates behavior

### ISSUE THREAD

**Issue by antmdvs** [jquense/react-formal#145]
Hi @jquense, thanks for this library. It's been working great for us so far.

It seems like when `alsoValidates` is specified and references elements that are invalid according to the schema, the element on which `alsoValidates` is specified fails validation, too, even if it's own value is valid.

To illustrate, consider the following schema:

```javascript
const validationSchema = yup.object({
  first: yup.string().max(5),
  last: yup.string().required(),
});
```

And these form fields:

```jsx
  <Form.Field name="first" alsoValidates="last" />
  <Form.Field name="last" />
```

If the user begins with a pristine form and types a character into the `first` input, then the `last` input becomes invalid (because it is `required()`. However, to my surprise, the `first` input ALSO becomes invalid even though it does not fail schema validation on its own.

You can see this here:

[![Edit nw7608prjp](https://codesandbox.io/static/img/play-codesandbox.svg)](https://codesandbox.io/s/nw7608prjp)

It seems like it's effectively failing `first` by association. Is this the intended behavior? If so, is there a way to suppress it? The `noValidate` prop wouldn't work since I DO want it to validate `max(5)`.

Thanks again!

Tony

**Comment by jquense** [comment]
looks like a bug, but the bug is really local to the `props.meta.invalid` property of the field. The form does not consider `first` actually invalid here (as evidenced by the message for the field not having anything to say). If you need to use `meta.invalid` or `valid` for this case the workaround would be to use `invalid = !!meta.errors[name].length`

**Comment by antmdvs** [comment]
Thanks, @jquense. That sounds like the opposite of #146.

**Comment by antmdvs** [comment]
Hey @jquense, has this been fixed? I updated react-formal to 0.28.1 in the sandbox and I'm no longer seeing this behavior.
